agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -457,13 +458,68 @@ bool slave_thread::handle_cmdu_control_ieee1905_1_message(Socket *sd,
     auto rx_mid = cmdu_rx.getMessageId();
 
     switch (cmdu_message_type) {
-    /* For Example:
+    case ieee1905_1::eMessageType::CHANNEL_PREFERENCE_QUERY_MESSAGE: {
 
-        case ieee1905_1::eMessageType::TOPOLOGY_DISCOVERY_MESSAGE :
-        {
-            break;
+        LOG(INFO) << "received CHANNEL_PREFERENCE_QUERY_MESSAGE";
+
+        // build channel preference report
+        auto cmdu_tx_header =
+            cmdu_tx.create(rx_mid, ieee1905_1::eMessageType::CHANNEL_PREFERENCE_REPORT_MESSAGE);
+        
+        if (!cmdu_tx_header) {
+            LOG(ERROR) << "cmdu creation of type CHANNEL_PREFERENCE_REPORT_MESSAGE, has failed";
+            return false;
+        }
+
+        auto channel_preference_tlv = cmdu_tx.addClass<wfa_map::tlvChannelPreference>();
+        if (!channel_preference_tlv) {
+            LOG(ERROR) << "addClass ieee1905_1::tlvChannelPreference has failed";
+            return false;
         }
-        */
+
+        channel_preference_tlv->radio_uid() =
+            network_utils::mac_1905_1_from_string(config.radio_identifier);
+
+        // Create operating class object
+        auto op_class_channels = channel_preference_tlv->create_operating_classes_list();
+        if (!op_class_channels) {
+            LOG(ERROR) << "create_operating_classes_list() has failed!";
+            return false;
+        }
+
+        // Fill operating class object
+        op_class_channels->operating_class() = 80; // random operating class for test purpose
+
+        // Fill up channels list in operating class object, with test values
+        for (uint8_t ch = 36; ch < 50; ch += 2) {
```

Comment:
This should be using BWL - we have now acs_report also in dummy mode, so use it.

Reply:
See here - https://github.com/prplfoundation/prplMesh/commit/cfcc9b21be1eef6dc4b667ffffb3d5e1ebd07688

Reply:
I understood from the task that what is needed it is create ping pong of dummy messages between the controller and the son_slave, only for certification purpose of the controller alone, without actually drill down the messages to the ap_manager/dwpal_hal.